nntrainer/tensor/weight.cpp

```diff
@@ -40,6 +42,24 @@ void Weight::runVariableInitializer() {
   Tensor &var_ref = getVariableRef();
   const TensorDim dim = var_ref.getDim();
 
+  unsigned int fan_in, fan_out;
+
+  /// @fixme: when unit is equal to one, this does not work, we need to rely on
+  /// effective dimension then actual numbers here. For now, some heuristics
+  /// added to infer what would be fan_in/fan_out
+  if (dim.batch() * dim.channel() * dim.height() == 1) {
+    fan_out = fan_in = dim.width();
```

Comment:
do we need this?? if batch == channel == height ==1, then field_size = dim.width() . then fan_in and fan_out = dim.width().

Reply:
Right :) I'll update.